Add Prometheus Configurator docs (NR-55439) (#9792)
newrelic/docs-website: src/content/docs/infrastructure/prometheus-integrations/install-configure-prometheus-configurator/scale-prometheus-instances-instances.mdx

---
title: Scaling Prometheus Agent instances
tags:
  - Integrations
  - Prometheus integrations
  - Install and configure Prometheus Configurator
metaDescription: TODO
---

> While our clusters keep growing in size, there will be more data gathered by Prometheus and at some point we can reach the maximum memory of the node; most of the time related to the cardinality of the time series.

Reviewer: ```suggestion
When clusters grow in size, there will be more data gathered by Prometheus and at some point it will reach the limits of how much data the Prometheus agent can process. The most common failure mode is running out of memory due to the increased cardinality of the time series.
```

Author: Thanks!

> In this situation you can observe that your Prometheus servers start dying (ideally this should be seen before) because they need more memory. This is an irrefutable sign that you need to face scaling.

Author: Alerts. But the alerting is up to the user. I am removing it.

class desc text;
class shards plain;
class targets plain;
```

### Target scraper identification
The shard identification (name of the StatefulSet Pod) is added as a `prometheus_server` label to all metrics and could be used to understand which is the Prometheus instance that is scraping that target.
To uniquely identify a Prometheus server instance within an account a combination of `cluster_name` and `prometheus_server` labels should be used.

### Self metrics

Usually, prometheus server self-metrics should be gathered from all prometheus servers, so the additional rules when sharding is configured should not apply to the job gathering the prometheus self-metrics. This is possible because the _configurator_ accepts the flag `skip_sharding` in the static_target jobs. This parameter is already set up in the default self-metrics job.

### Limitations

If additional scrape jobs are included in configuration as `extra_scrape_configs`, as that field will hold the raw definition of prometheus jobs, the _configurator_ will not include the rules corresponding to sharding configuration and, as a result, the corresponding targets will be scrapped by all
prometheus servers.

> Currently, auto-scaling is not supported. If increasing or decreasing the number of shards is required, an update of the cluster settings (which implies restarting the prometheus servers) is needed.

Reviewer: ```suggestion
Currently, auto-scaling is not supported. To increase or decrease the number of shards, you will need to an update of the chart settings. This will restart the prometheus pods.
```

Author: Thanks!
